akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessState.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2009-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.LoggerOps
+import akka.cluster.ddata.LWWRegister
+import akka.cluster.ddata.LWWRegisterKey
+import akka.cluster.ddata.typed.scaladsl.DistributedData
+import akka.cluster.ddata.typed.scaladsl.Replicator
+import akka.cluster.sharding.typed.ShardedDaemonProcessContext
+
+import java.time.Instant
+
+/**
+ * INTERNAL API
+ */
+private[akka] final case class ShardedDaemonProcessState(
+    revision: Long,
+    numberOfProcesses: Int,
+    completed: Boolean,
+    started: Instant)
```

Comment:
What do you think about making this a proper `ReplicatedData`? Merge should be dead simple since the revision is increasing and completed is changing from false to true.

Reply:
I don't have a great motivation but I don't think I ever did a custom ReplicatedData structure so I'll do it.

Reply:
Changed in [d530fe8](https://github.com/akka/akka/pull/31844/commits/d530fe89753170bd74809cc4d7982d73a04deb96)